### PR TITLE
File handling

### PR DESCRIPTION
**Pull request by mas8074**

### CODE REVIEW

File: src/file_input/MIDI_file_class.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+    
+    def get_next_message(self):
+        """Iterate current position and get the next message.
+        This can be called iteratively.
+
+        Returns:
+            dict: dictionary representation of next message
+        """
+        if self.curr_pos < len(self.messages):
```
Comment (cha2251) [inline src/file_input/MIDI_file_class.py:87]: I think there is an off by one here:
`if self.curr_pos < len(self.messages)-1:`

If list length is 3 and our current is 2, we would pass the check. But then would increment current to three and try to grab the message at position 3.

File: test/file_input_tests/MIDI_file_class_test.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
+
+import src.file_input.MIDI_file_class as MIDI_FC
+
+class TestParse:
+
+    def test_parse_midi_file(self):
+        """ This is the test to ensure the function returns the correct list of midi messages.
+        TODO: complete this test, test with smaller file
```
Comment (cha2251) [inline test/file_input_tests/MIDI_file_class_test.py:8]: Fine with leaving this incomplete for now, but make sure to put something about this in Trello so it doesn't get lost

File: test/file_input_tests/MIDI_file_class_test.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        expected = []
+        actual = val
+
+        assert actual == expected
+
+    
+    def test_get_next_message(self):
+        """
+        
+        {'type': 'track_name', 'name': 'Piano', 'time': 0}
```
Comment (cha2251) [inline test/file_input_tests/MIDI_file_class_test.py:33]: Quick description on what the values are in this comment would be good. Took me a sec

**Review by cha2251 — CHANGES_REQUESTED** [review]
Couple small things, but this looks great

File: src/file_input/MIDI_file_class.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+    # iterate over midi file
+    for i, track in enumerate(mid_fi.tracks):
+        print('Track {}: {}'.format(i, track.name))
+
+        # iterate over messages in each track
+        if track.name == track_string:
+            for msg in track:
+                track_messages.append(msg.dict())
+                print(msg.dict())
```
[the remark this answers is not in the document]
Comment (mas8074) [reply src/file_input/MIDI_file_class.py:34]: Of course, of course. Thanks for the reminder! I forgot to do that

File: src/file_input/MIDI_file_class.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            
+    return track_messages
+
+def is_file_string_valid(file_string):
+
+    is_valid = False
+    if file_string != '':
```
[the remark this answers is not in the document]
Comment (mas8074) [reply src/file_input/MIDI_file_class.py:41]: That's probably a good idea lol

File: src/file_input/MIDI_file_class.py [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+        is_valid = True
+    
+    return is_valid
+
+
+
+class MIDIFileObject:
```
[the remark this answers is not in the document]
Comment (mas8074) [reply src/file_input/MIDI_file_class.py:48]: I want to know what you mean by "move it to a separate file" if I'm just going to move the function inside the class?

[the remark this answers is not in the document]
Comment (mas8074) [reply src/file_input/MIDI_file_class.py:48]: That's totally fair, I just moved it inside the class.

File: src/file_input/MIDI_file_class.py [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    def __str__(self):
+        """String representation of object
+
+        Returns:
+            str: printable string
+        """
+        return f"{self.file_name}"
+
+    
+    def get_next_message(self):
```
[the remark this answers is not in the document]
Comment (mas8074) [reply src/file_input/MIDI_file_class.py:80]: I like that solution, I can explain why I ended up doing it this way in our next meeting, but you make perfect sense.

File: src/file_input/MIDI_file_class.py (hunk 1 above)
Comment (mas8074) [reply src/file_input/MIDI_file_class.py:87]: The way it is currently implemented:

File: test/file_input_tests/MIDI_file_class_test.py (hunk 2 above)
Comment (mas8074) [reply test/file_input_tests/MIDI_file_class_test.py:8]: Good idea, my thinking is that we will go back during our testing phase and I can test it.

File: test/file_input_tests/MIDI_file_class_test.py (hunk 3 above)
Comment (mas8074) [reply test/file_input_tests/MIDI_file_class_test.py:33]: Sure

**Review by cha2251 — APPROVED** [review]
lgtm

File: src/file_input/MIDI_file_class.py (hunk 1 above)
Comment (cha2251) [reply src/file_input/MIDI_file_class.py]: resolved offline, fixed out of bounds

**Pull request merged**